react/src/index.tsx

```diff
@@ -29,43 +39,54 @@ interface SchematicHookOpts {
 
 type UseSchematicFlagOpts = SchematicHookOpts & {
   fallback?: boolean;
-}
+};
 
 const SchematicContext = createContext<SchematicContextProps>({
   flagValues: {},
 });
 
-
 const SchematicProvider: React.FC<SchematicProviderProps> = ({
   children,
   client: providedClient,
-  clientOpts,
   publishableKey,
+  ...clientOpts
 }) => {
-  const [client, setClient] = useState<SchematicJS.Schematic | undefined>();
+  const [client, setClient] = useState<SchematicJS.Schematic>();
   const [flagValues, setFlagValues] = useState<Record<string, boolean>>({});
+  /* eslint-disable-next-line react-hooks/exhaustive-deps */
+  const memoizedClientOpts = useMemo(() => clientOpts, [JSON.stringify(clientOpts)]);
```

Comment:
Maybe there is a better way to do this? I had to add this to prevent infinite rerendering

Reply:
We can always go back and optimize later, but if it works with `JSON.stringify` let's keep it for now.